### PR TITLE
PT-3594: Don't append a free text option if input has an exact match in the suggestion list

### PR DESCRIPTION
**Pull request by veronikaslc**

### CODE REVIEW

File: components/widgets/ui/src/main/resources/PhenoTips/Widgets.xml [hunk 1]
```diff
@@ -1194,8 +1194,15 @@
     // loop throught arr of suggestions
     // creating an XlistItem for each suggestion
     //
+    // at the same time check if any of the suggestions
+    // is the exact case-sentitive match of the input
```
Comment (sdumitriu) [inline components/widgets/ui/src/main/resources/PhenoTips/Widgets.xml]: case-insensitive

Comment (veronikaslc) [reply components/widgets/ui/src/main/resources/PhenoTips/Widgets.xml]: right, that's what I actually meant :)

File: components/widgets/ui/src/main/resources/PhenoTips/Widgets.xml [hunk 2]
```diff
@@ -1194,8 +1194,15 @@
...
+    //
+    var exactMatch = false;
     for (var i = 0,len = arr.length; i &lt; len; i++)
     {
+       if (arr[i].value.toLowerCase() == this.fld.value.strip().toLowerCase()) {
```
Comment (sdumitriu) [inline components/widgets/ui/src/main/resources/PhenoTips/Widgets.xml]: This should also look in `arr[i].id`

**Comment by sashaandjic** [comment]
tested in master

**Pull request merged**